src/_drawLegend.js

```diff
@@ -57,28 +63,25 @@ export default function(data = []) {
       .key(fill)
       .rollup(leaves => legendData.push(merge(leaves, this._aggs)))
       .entries(this._colorScale ? data.filter((d, i) => this._colorScale(d, i) === undefined) : data);
-
+    
     this._legendClass
       .id(fill)
       .align(wide ? "center" : position)
       .direction(wide ? "row" : "column")
       .duration(this._duration)
       .data(legendData.length > 1 || this._colorScale ? legendData : [])
-      .height(this._height - this._margin.bottom - this._margin.top)
+      .height(wide ? this._height - (this._margin.bottom + this._margin.top) : this._height - (this._margin.bottom + this._margin.top + this._legendMargin.bottom + this._legendMargin.top))
       .select(legendGroup)
       .verticalAlign(!wide ? "middle" : position)
-      .width(this._width - this._margin.left - this._margin.right)
+      .width(wide ? this._width - (this._margin.left + this._margin.right + this._legendMargin.left + this._legendMargin.right) : this._width - (this._margin.left + this._margin.right))
```

Comment:
I don't think these `width` and `height` declaractions need the ternary? I believe we should just always be subtracting the margin and the padding.